### PR TITLE
Switch to go modules and update k8s code to 1.17.0

### PR DESCRIPTION
**Pull request by cofyc**
This PR does

- upgrade go to 1.13.9
- switch to go modules
- update k8s code to 1.17.0 (1.18.0 changed client interface, we can upgrade later)

List of commits:

- ~~048603b86f181eca7d0cc7030115ebc11cc758e0, 295a3a41bd7fccbc1830390b48df4bea10970653: add repo-infra with  `git subtree` (`repo-infra` cannot be vendored by `go mod`)~~
- ce8f17d6ee60b3787150723ff787273d8a8f6cde: vendor changes
- bb5c95057c69f72b66e4b6719c6940bf21787a5f: non-vendor changes

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,73 @@
+module sigs.k8s.io/sig-storage-local-static-provisioner
+
+go 1.14
```
Comment (msau42) [inline go.mod:3]: Should this be 1.13?

Comment (cofyc) [reply go.mod]: reverted

I originally wanted to upgrade go to 1.14, but the prow image uses 1.13.

**Comment by cofyc** [comment]
`repo-infra` contains owners that are not members of `kubernetes-sigs` org. Forge merge or copy `repo-infra/hack/verify_boilerplate.p` file into this repo?

**Comment by msau42** [comment]
Also I'm not sure if we can merge with the owners file issue.

Must we have these repo-infra utilities?

**Comment by msau42** [comment]
I am fine with copying the file over

**Comment by cofyc** [comment]
> I am fine with copying the file over

done

**Pull request merged**